kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/KubernetesResourceUtil.java

```diff
@@ -423,14 +424,46 @@ private static Class<?> loadRelated(Class<?> type, String suffix, Class<?> defau
 
   /**
    * Create Secret by using username and password.
-   * 
+   *
    * @param dockerServer User to store key value pair for auths map
    * @param username username that needs to be used during secret creation
    * @param password password that needs to be used during secret creation
    * @return an object of Secret
    */
   public static Secret createDockerRegistrySecret(String dockerServer, String username, String password)
       throws JsonProcessingException {
+    Map<String, Object> dockerConfigMap = createDockerRegistryConfigMap(dockerServer, username, password);
+    String dockerConfigAsStr = Serialization.jsonMapper().writeValueAsString(dockerConfigMap);
+
+    return new SecretBuilder()
+        .withNewMetadata().withName(DEFAULT_DOCKER_REGISTRY_SECRET_NAME).endMetadata()
+        .withType("kubernetes.io/dockerconfigjson")
+        .addToData(".dockerconfigjson", Base64.getEncoder().encodeToString(dockerConfigAsStr.getBytes(StandardCharsets.UTF_8)))
+        .build();
```

Comment:
OK but now you have code duplication, here I would simply call 
```suggestion
    return createDockerRegistrySecret(dockerServer, username, password, DEFAULT_DOCKER_REGISTRY_SECRET_NAME);
```

Reply:
also, where is `createDockerRegistrySecret` called?